app/lib/search/backend.dart

```diff
@@ -158,29 +158,34 @@ class SearchBackend {
     }
 
     // initial scan of packages
-    final pool = Pool(concurrency);
-    final futures = <Future>[];
-    await for (final package in dbService.query<Package>().run()) {
-      if (package.isNotVisible) {
-        continue;
+    await withVmTerminationTimeout(timeout: Duration(hours: 12), () async {
```

Comment:
Why not just add `.timeout(Duration(hours: 12))` when calling `doCreateAndUpdateSnapshot`?

If such a timeout does happen there, isn't that a good place to just not catch the timeout and let it bubble all the way up? Do we need to explicitly kill the VM?